src/io/estrado/Pipeline.groovy

```diff
@@ -5,7 +5,33 @@ def kubectlTest() {
     // Test that kubectl can correctly communication with the Kubernetes API
     println "checking kubectl connnectivity to the API"
     sh "kubectl get nodes"
+}
+
+def buildDockerImage(appName, buildId) {
+    stage('Build Docker Image') {
+        container('docker') {
+            sh "docker build -t vemba/${appName}:${buildId} ."
+            echo "Image built"
+        }
+    }
+}
+
+def getPodLabel(cloud) {
+    return "pipeline-$cloud-${UUID.randomUUID().toString()}".take(62)
+}
 
+def getCloud(branch) {
+    if (branch == 'master' ) {
+        cloud = "kubernetes"
+    } else if (branch == 'staging' ) {
+        cloud = "kubernetes-staging"
+    } else if (branch == 'staging-new' ) {
+        cloud = "kubernetes-staging-new"
```

Comment:
i thought staging-new and k8s-dev don't exist anymore?

Reply:
I will create next pr once I merge this one.